src/mono/System.Private.CoreLib/src/System/Runtime/DependentHandle.cs

```diff
@@ -0,0 +1,170 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime
+{
+    // The ephemeron type is checked in the Mono runtime in "object.c" as follows:
+    //
+    // m_class_get_image (klass) == mono_defaults.corlib && !strcmp ("Ephemeron", m_class_get_name (klass))
+    //
+    // As such, the namespace it belongs to in the managed runtime doesn't matter: the VM will just check
+    // that the type name matches, and that the type is in fact defined in corelib. This means we can
+    // just move it to System.Runtime without worrying about it being properly managed in the VM.
+    // Additionally, the type is defined in "sgen-mono.c" as follows:
+    //
+    // typedef struct {
+    //     GCObject* key;
+    //     GCObject* value;
+    // } Ephemeron;
+    //
+    // So as long as the layout matches the one of the type defined in C# (which it does), we're also free
+    // to rename the fields to better follow the naming guidelines, and the VM will have no issues with it.
+    internal struct Ephemeron
+    {
+        public object? Key;
+        public object? Value;
+    }
+
+    //
+    // Instead of dependent handles, mono uses arrays of Ephemeron objects.
+    //
+    public struct DependentHandle : IDisposable
+    {
+        private Ephemeron[]? _data;
+
+        public DependentHandle(object? target, object? dependent)
+        {
+            _data = new Ephemeron[1];
+            _data[0].Key = target;
+            _data[0].Value = dependent;
+            GC.register_ephemeron_array(_data);
+        }
+
+        public bool IsAllocated => _data is not null;
+
+        public object? Target
+        {
+            get => GetTarget();
+            set => SetTarget(value);
+        }
+
+        public object? Dependent
+        {
+            get => GetDependent();
+            set => SetDependent(value);
+        }
+
+        public (object? Target, object? Dependent) GetTargetAndDependent()
+        {
+            if (_data is not Ephemeron[] data)
+            {
+                ThrowHelper.ThrowInvalidOperationException();
+
+                return default;
+            }
+
+            Ephemeron e = data[0];
+
+            return e.Key != GC.EPHEMERON_TOMBSTONE ? (e.Key, e.Value) : default;
+        }
+
+        internal object? UnsafeGetTarget()
+        {
+            return GetTarget();
```

Comment:
`GetTarget()` can be here, and all places that call `GetTarget` can call `UnsafeGetTarget` instead. No need to have it in a separate method.

Reply:
Done in 0f6f3d50ebf2abfd094ab3678128f40cb328c19b for all 3 applicable methods 👍